src/metabase/api/collection.clj

```diff
@@ -54,29 +85,30 @@
   [archived namespace]
   {archived  (s/maybe su/BooleanString)
    namespace (s/maybe su/NonBlankString)}
-  (let [archived? (Boolean/parseBoolean archived)]
-    (as-> (db/select Collection
-            {:where    [:and
-                        [:= :archived archived?]
-                        [:= :namespace namespace]
-                        (collection/visible-collection-ids->honeysql-filter-clause
-                         :id
-                         (collection/permissions-set->visible-collection-ids @api/*current-user-permissions-set*))]
-             :order-by [[:%lower.name :asc]]}) collections
-      ;; include Root Collection at beginning or results if archived isn't `true`
-      (if archived?
-        collections
-        (let [root (root-collection namespace)]
-          (cond->> collections
-            (mi/can-read? root)
-            (cons root))))
-      (hydrate collections :can_write)
-      ;; remove the :metabase.models.collection.root/is-root? tag since FE doesn't need it
-      ;; and for personal collections we translate the name to user's locale
-      (for [collection collections]
-        (-> collection
-            (dissoc ::collection.root/is-root?)
-            collection/personal-collection-with-ui-details)))))
+  (let [archived? (Boolean/parseBoolean archived)
+        q         (coll-query
+                   api/*current-user-id*
+                   [:= :c.archived archived?]
+                   [:= :c.namespace namespace]
+                   (collection/visible-collection-ids->honeysql-filter-clause
+                    :c.id
+                    (collection/permissions-set->visible-collection-ids @api/*current-user-permissions-set*)))]
```

Comment:
I see both calls to `coll-query` are passing in these and values. Is there a good reason not to just include it inside the function instead of passing it in?

Reply:
Yeah, went back and forth on that. The `namespace` arg is common across both uses, but the `archived?` flag is not. The query itself is easier to debug if we keep it out of `coll-query` since local REPLers won't need to get or mock permissions. I can go either way. Anyone have a strong opinion on this one?